src/FeedsMigrateImporterInterface.php

```diff
@@ -19,4 +19,155 @@ interface FeedsMigrateImporterInterface extends ConfigEntityInterface {
+   * @param string $existing
+   *   One of the following:
+   *   - 'skip' => FeedsMigrateImporterInterface::EXISTING_SKIP.
+   *   - 'replace' => FeedsMigrateImporterInterface::EXISTING_REPLACE.
+   *   - 'update' => FeedsMigrateImporterInterface::EXISTING_UPDATE.
+   */
+  public function setExisting(string $existing);
+
+  /**
+   * Returns if orphaned items should be kept.
+   *
+   * @return bool
+   *   TRUE if orphans should be kept, FALSE if they can be deleted.
+   */
+  public function keepOrphans();
+
+  /**
+   * Sets if orphaned items should be kept.
+   *
+   * @param bool $keep_orphans
+   *   TRUE if orphans should be kept, FALSE if they can be deleted.
+   */
+  public function setKeepOrphans(bool $keep_orphans);
+
+  /**
+   * Get the timestamp of the last import.
+   *
+   * @return int
+   *   Unix timestamp.
+   */
+  public function getLastRun();
+
+  /**
+   * Update the timestamp during which the import last ran.
+   *
+   * @param int $last_run
+   *   Unix timestamp of the last import.
+   */
+  public function setLastRun(int $last_run);
+
+  /**
+   * Get the id of the migration.
+   *
+   * @return string
+   *   ID of the migration plugin.
+   */
+  public function getMigrationId();
+
+  /**
+   * Update the timestamp during which the import last ran.
+   *
+   * @param string $id
+   *   ID of the migration plugin.
+   */
+  public function setMigrationId(string $id);
+
+  /**
+   * Get the original migration plugin.
+   *
+   * @return \Drupal\migrate_plus\Entity\MigrationInterface
+   *   The original migration entity before any configuration changes.
+   */
+  public function getOriginalMigration();
+
+  /**
+   * Get the altered migration plugin.
+   *
+   * @return \Drupal\migrate_plus\Entity\MigrationInterface
+   *   The migration plugin after configuration changes are applied.
+   */
+  public function getMigration();
+
+  /**
+   * If the import should be executed.
+   *
+   * @return bool
+   *   TRUE if it should be run on cron, FALSE otherwise.
+   */
+  public function needsImport();
+
+  /**
+   * Get the altered migrate batch executable object that can run the import.
+   *
+   * @return \Drupal\feeds_migrate\FeedsMigrateExecutable
+   *   The batch executable to run the import.
+   *
+   * @throws \Drupal\migrate\MigrateException
+   *   If the executable failed.
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   *   If the instance cannot be created, such as if the ID is invalid.
+   */
+  public function getBatchExecutable();
+
+  /**
+   * Get the altered migrate executable object that can run the import.
+   *
+   * @return \Drupal\feeds_migrate\FeedsMigrateExecutable
+   *   The executable to run the import.
+   *
+   * @throws \Drupal\migrate\MigrateException
+   *   If the executable failed.
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   *   If the instance cannot be created, such as if the ID is invalid.
+   */
+  public function getExecutable();
+
 }
```

Comment:
Nice. I see that also PHP 7 type hints are used which is also nice. They are not part of my routine yet as PHP 5 is still somewhat supported. But in a few months PHP 5 is out of support for Drupal core, so no reason to support that in Feeds Migrate.